ariadne/src/Ariadne/Config/Presence.hs

```diff
@@ -0,0 +1,34 @@
+module Ariadne.Config.Presence
+  ( Presence (There, File)
+  , _File
+  ) where
+
+import Universum
+
+import Control.Lens (Prism', prism')
+
+data Presence a
+  = There a
+  | File (Maybe FilePath)
+  deriving (Eq, Show)
+
+instance Functor Presence where
```

Comment:
Can be derived using `DeriveFunctor`.

Reply:
What for embed a file into memory if we have can just use default configuration? `There` is for this needs.

Reply:
How is it related to my suggestion to use `DeriveFunctor` instead of writing this instance manually? :thinking:

Reply:
I like to write instances myself, it's one minute, but I understand what happens. The question about `There` is not related to this, but I don't know where I can answer on your question above. The goal of the task was to avoid using filepaths for default configurations, instead, we can define our default configuration values. But it can be used unless Cardano configuration types will be ripped out.

Reply:
You can write comments directly to this PR quoting the parts you are answering to.
I've written about the rest in this thread: https://serokell.slack.com/archives/C9B1DAMDJ/p1531476663000117